lib/ansiblelint/utils.py

```diff
@@ -732,3 +739,108 @@ def get_rule_skips_from_line(line):
         noqa_text = line.split('# noqa')[1]
         rule_id_list = noqa_text.split()
     return rule_id_list
+
+
+def is_playbook(filename):
+    """
+    Given a filename, it should return true if it looks like a playbook. The
+    function is not supposed to raise exceptions.
+    """
+    # we assume is a playbook if we loaded a sequence of dictionaries where
+    # at least one of these keys is present:
+    playbooks_keys = {
+        "gather_facts",
+        "hosts",
+        "import_playbook",
+        "post_tasks",
+        "pre_tasks",
+        "roles"
+        "tasks",
+    }
+
+    # makes it work with Path objects by converting them to strings
+    if not isinstance(filename, six.string_types):
+        filename = str(filename)
+
+    try:
+        f = parse_yaml_from_file(filename)
+        if isinstance(f, AnsibleSequence) and playbooks_keys.intersection(f[0].keys()):
+            return True
+    except Exception as e:
+        print(
+            "Warning: Failed to load %s with %s, assuming is not a playbook."
+            % (filename, e))
+    return False
+
+
+def normpath(path):
+    """
+    Normalize a path in order to provide a more consistent output. Currently
+    this means to generate a relative path but in the future we may want to
+    make this configurable.
+    """
+    return os.path.relpath(path)
+
+
+def get_playbooks_and_roles(options={}):
+
+    # git is preferred as it also considers .gitignore
+    files = OrderedDict.fromkeys(sorted(subprocess.check_output(
+        ["git", "ls-files", "*.yaml", "*.yml"],
+        universal_newlines=True).split()))
+
+    playbooks = []
+    role_dirs = []
+    role_internals = {
+        'defaults',
+        'files',
+        'handlers',
+        'meta',
+        'tasks',
+        'templates',
+        'vars',
+        }
+
+    # detect role in repository root:
+    if 'tasks/main.yml' in files:
+        role_dirs.append('.')
+
+    for p in map(Path, files):
+
+        if any(str(p).startswith(file_path) for file_path in options.exclude_paths):
+            continue
+        elif next((i for i in p.parts if i.endswith('playbooks')), None) \
```

Comment:
Please use braces instead of escaping EOL

Reply:
@webknjaz I had impression that code-formatting was sometime to be avoided in reviews. That is why we have linters and formatters to take care of these. Discussions about style are a waste of human time. I will do the change but lets find a way to avoid them in the first place.